File: formal/coqimp/task05/extraction_ml/Archi.ml
open BinNums
open Datatypes
open Fappli_IEEE

(** val default_pl_64 : bool * nan_pl **)

let default_pl_64 =
  (true,
    (let rec f = function
     | O -> Coq_xH
     | S n0 -> Coq_xO (f n0)
     in f (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S
          (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S
          (S (S (S (S (S (S (S
          O)))))))))))))))))))))))))))))))))))))))))))))))))))))

(** val choose_binop_pl_64 : bool -> nan_pl -> bool -> nan_pl -> bool **)

let choose_binop_pl_64 _ _ _ _ =
  false

File: formal/coqimp/task05/extraction_hs/Archi.hs
module Archi where

import qualified Prelude
import qualified Datatypes
import qualified Fappli_IEEE

default_pl_64 :: (,) Prelude.Bool Fappli_IEEE.Coq_nan_pl
default_pl_64 =
  (,) Prelude.True
    (Datatypes.nat_rect 1 (\_ x -> (\x -> 2 Prelude.* x) x) (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      0))))))))))))))))))))))))))))))))))))))))))))))))))))

choose_binop_pl_64 :: Prelude.Bool -> Fappli_IEEE.Coq_nan_pl -> Prelude.Bool ->
                      Fappli_IEEE.Coq_nan_pl -> Prelude.Bool
choose_binop_pl_64 _ _ _ _ =
  Prelude.False

